src/AbstractObjectCollection.php

```diff
@@ -69,6 +70,16 @@ public function __construct(
         return parent::offsetGet($this->createValidIndex($index));
     }
 
+    final public function natcasesort(): void
+    {
+        throw new LogicException('natcasesort is not supported for objects.');
```

Comment:
These methods should not be disabled, cause they will work fine if the object has the __toString method.
I did not create an explicit check for this, because the default "Object of class Foo could not be converted to string" is a common default exception for this. 

The documentation/functionality of the natcasesort and natsort, which use the array values in their string algorithms, should points this out as well.

Reply:
Removed it